src/ui/common/src/components/pages/workflows/index.tsx

```diff
@@ -125,14 +126,36 @@ const WorkflowsPage: React.FC<Props> = ({ user, Layout = DefaultLayout }) => {
         user.apiKey,
         workflowId
       );
-      let status = ExecutionStatus.Unknown;
 
-      if (latestDagResult) {
-        status = latestDagResult.exec_state.status;
-      } else if (dag) {
-        status = ExecutionStatus.Registered;
-      }
-      return <ExecutionStatusLink name={row.name} url={url} status={status} />;
+      const latestDagResultId = latestDagResult?.id;
+      const latestDagId = latestDagResult?.dag_id ?? dag?.id;
+      const workflowStatus = latestDagResult?.exec_state?.status;
+
+      const nodes = useWorkflowNodes(user.apiKey, workflowId, latestDagId);
+      const nodesResults = useWorkflowNodesResults(
+        user.apiKey,
+        workflowId,
+        latestDagResultId
+      );
+
+      const ready =
+        Object.getOwnPropertyNames(nodes.operators).length !== 0 ? true : false;
```

Comment:
What does `getOwnPropertyNames` do? Since we treat this as a map it would be more clear to use `Object.keys()`. Also, you don't need the `? :` sugar here and can directly do `const ready = ().length !== 0`.